File: app/src/main/java/com/future/awaker/db/dao/BannerDao.java
package com.future.awaker.db.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.future.awaker.data.BannerItem;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface BannerDao {

    @Query("SELECT * FROM banner order by id desc")
    Flowable<List<BannerItem>> loadAllBanners();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllBanners(List<BannerItem> bannerItems);
}

File: article/src/main/java/com/ruzhan/awaker/article/db/dao/BannerDao.kt
package com.ruzhan.awaker.article.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query

import com.ruzhan.awaker.article.model.BannerItem

import io.reactivex.Flowable


@Dao
interface BannerDao {

    @Query("SELECT * FROM banner order by id desc")
    fun loadAllBanners(): Flowable<List<BannerItem>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllBanners(bannerItems: List<BannerItem>)
}
